lib/nearley.js

```diff
@@ -234,12 +234,19 @@
         }
         this.grammar = grammar;
 
+        if (!options) {
+            options = {}
+        }
+
         // Read options
         this.options = {
             keepHistory: false,
-            lexer: grammar.lexer || new StreamLexer,
+            lexer: options.synchronous ?
+                (grammar.lexer ? grammar.lexer.clone() : new StreamLexer) :
+                (grammar.lexer || new StreamLexer),
```

Comment:
I propose to change the statement to avoid repetition:
```javascript
...
lexer: grammar.lexer ? (options.synchronous ? grammar.lexer.clone() : grammar.lexer) : new StreamLexer,
...
```

Reply:
That's a good idea, I'll change it now in the pull request

Reply:
Done